### PR TITLE
rollout CRD API Spec

### PR DESCRIPTION
**Pull request by ryanzhang-oss**

### CODE REVIEW

File: apis/standard.oam.dev/v1alpha1/rollout_types.go [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+	// Size is the number of pods to upgrade in this batch
+	// it can be an absolute number (ex: 5) or a percentage of total pods
+	Size intstr.IntOrString `json:"size"`
+
+	// MaxUnavailable is the max allowed number of pods that is unavailable
+	// during the upgrade. We will mark the batch as ready as long as there are less
+	// or equal number of pods unavailable than this number.
+	// default = 0
+	// +optional
+	MaxUnavailable *intstr.IntOrString `json:"batchInterval,omitempty"`
```
Comment (zzxwill) [inline apis/standard.oam.dev/v1alpha1/rollout_types.go:121]:
```suggestion
	MaxUnavailable *intstr.IntOrString `json:"maxUnavailable,omitempty"`
```

File: apis/standard.oam.dev/v1alpha1/rollout_types.go [hunk 2]
```diff
@@ -109,16 +115,23 @@ type RolloutSpec struct {
...
 	// it can be an absolute number (ex: 5) or a percentage of total pods
-	Size intstr.IntOrString `json:"size"`
+	// +optional
+	// it is mutually exclusive with the PodList field
+	Replica intstr.IntOrString `json:"replica,omitempty"`
+
+	// The list of Pods to get upgraded
+	// +optional
+	// it is mutually exclusive with the Replica field
+	PodList []string `json:"podList,omitempty"`
```
Comment (zzxwill) [inline apis/standard.oam.dev/v1alpha1/rollout_types.go:127]: Is it the list of pod names?

Comment (ryanzhang-oss) [reply apis/standard.oam.dev/v1alpha1/rollout_types.go:127]: Yes, any suggestions on the field name?

File: docs/examples/testapp/vela.yaml [hunk 3]
```diff
@@ -3,7 +3,7 @@ name: testapp
 services:
   express-server:
     # this image will be used in both build and deploy steps
-    image: oamdev/testapp:v1
+    image: nerdyyatrice/testapp:v1
```
Comment (wonderflow) [inline docs/examples/testapp/vela.yaml:6]: change it back?

File: docs/examples/testapp/vela.yaml [hunk 4]
```diff
@@ -23,11 +23,11 @@ services:
...
-    # route:
-    #   domain: example.com
-    #   rules:
-    #     - path: /testapp
-    #       rewriteTarget: /
+    route:
+      domain: example.com
+      rules:
+        - path: /testapp
+          rewriteTarget: /
```
Comment (wonderflow) [inline docs/examples/testapp/vela.yaml:30]: change back?

**Review by wonderflow — APPROVED** [review]
Generally LGTM.

**Comment by wonderflow** [comment]
please rebase

**Comment by hongchaodeng** [comment]
@ryanzhang-oss Could you provide a top down design under [design/](https://github.com/oam-dev/kubevela/tree/master/design/vela-core) or write up an issue to describe the whole picture first? I couldn't find any design or issue for this yet.

**Comment by wonderflow** [comment]
The spec generally LGTM, let's merge this one and discuss in #892. We can change the spec in that PR if we feel the design need to change anywhere.

**Pull request merged**